### ISSUE TITLE
Enhance first party support for mods

### ISSUE THREAD

**Issue by gentlecolts** [Yellow-Dog-Man/Resonite-Issues#812]
### Is your feature request related to a problem? Please describe.

Working with mods is currently quite cumbersome, and relies heavily on grabbing random files off scattered locations in github, keeping those files updated, and, in the case of bugs, triaging whether or not a mod has broken or the bug is an actual froox engine issue.

### Describe the solution you'd like

I believe first party systems could be put in place to more cleanly support modding.  I would request the addition of a "mods" tab on the dashboard.  This dashboard page should meet the following requirements:
- [ ] All installed mods should be made visible
- [ ] Clear indication that the Resonite team is not responsible for defects caused by mods, and any other similar disclaimers
- [ ] Provide a means of checking for mod updates.  Mods should be responsible for telling this dashboard where to look, subject to the same url access prompt as websockets
- [ ] Ability to enable/disable mods from this dashboard.  Ideally this would happen live, however it is acceptable if this requires a restart to take effect.
- [ ] If an uncaught exception is raised from a mod, it should be disabled automatically (ideally without crash) until the user manually re-enables it.  It should be indicated to the user that this is why the mod was disabled.  I suspect this feature in particular to be very important in reducing developer strain from mod-related bug reports.
- [ ] Ability to install new mods via the UI.  There are many ways this could be accomplished, however I would suggest that mod makers provide a plaintext config file of some kind, which users can download and provide to resonite through this interface.  This config would be responsible for indicating where to download (and get updates for) this mod, among any other necessary information (name, version, author, etc).  Some Resonite-hosted mod repository would be nice, but not required
- [ ] Provide an interface for which mods can display their own settings within this page.  Ex: mods in a vertical list on the left, the currently selected mod's settings displayed on the right

### Describe alternatives you've considered

This is not a small request, but my goal is to provided a more comfortable mod experience without requiring the team to become too responsible for dealing with mods themselves.  I suspect this ticket to be very open-ended, and would greatly appreciate more feedback from devs and mod makers

### Additional Context

zangooseoo

**Comment by XDelta** [comment]
Several of these points already exist although not as first party things. 
- The ModSettings mod can show all the installed mods, regardless if they have settings
- There are some disclaimers around mods, it is also mentioned on the this repo readme about not running plugins or mods when reporting issues. 
- For updating and managing mods, tools are currently being worked on for doing so. 
- Enabling/disabling a given mod while playing has to be supported by a given mod, although many do provide that option. 
- Several cases where a mod causes breakage will be caught and logged. When possible the mod just won't continue loading while letting the rest continue to work. Its difficult with the nature of how they work to isolate them further and any breakage might not be able to be automatically determined.
- Again, will be a job more meant for tools outside of the game to install and update mods
- There is a mod to show settings for other mods in-game https://github.com/badhaloninja/ResoniteModSettings

**Comment by gentlecolts** [comment]
Yep!  Thanks for calling those out explicitly, part of my inspiration for this post WAS existing mods that provide this functionality, and bringing that functionality to first-party.  "a mod to make the mods work better" is still a mod, and still has the problem of not being first party, in particular being more at risk of breakage in response to first-party updates, and being harder to discover in the first place

**Comment by Geenz** [comment]
Unfortunately, we’re not going to be enhancing or extending first party support for mods. We could create enhancements for plugins - but those should be raised as separate issues.

**Comment by XDelta** [comment]
Mods themself aren't even an officially supported feature, the modloader uses the plugin system which is officially supported so it may be better to direct requested enhancements to the supported plugin feature

**Comment by Frooxius** [comment]
Adding a bit more context: We do not have first party support for mods in the first place - they are using systems that are outside of our own - it's not something we can enhance.

We do have plans to rework our core data model, which will provide some improvements and allow plugins not break compatibility between versions: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/702
